src/Microsoft.AspNet.Mvc.TagHelpers/InputTagHelper.cs

```diff
@@ -43,8 +43,6 @@ public class InputTagHelper : TagHelper
                 { nameof(UInt32), "number" },
                 { nameof(Int64), "number" },
                 { nameof(UInt64), "number" },
-                { nameof(Single), "number" },
-                { nameof(Double), "number" },
                 { nameof(Boolean), InputType.CheckBox.ToString().ToLowerInvariant() },
                 { nameof(Decimal), InputType.Text.ToString().ToLowerInvariant() },
```

Comment:
Please also remove this line because the template is not `decimal`-specific and is now inconsistent with how `float` and `double` reach `type="text"`.

Alternatively restore the `float` and `double` lines with value `InputType.Text.ToString().ToLowerInvariant()`. This alternative will behave no differently (well, it'll save one lookup iteration) but will make the table appear even-more different from that in `TemplateRenderer`. Of course that table _is_ different because we have a `decimal`-specific template there.